File: backend/ERP.Backend/Domain/Interfaces/IUseCases/IEstadoPedidoUseCases.cs
using Application.DTOs;
using Domain.DTOs;
using Domain.Entities;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Domain.Interfaces.IUseCases
{
    /// <summary>
    /// Interfaz de los casos de uso de estados de pedido.
    /// </summary>
    public interface IEstadoPedidoUseCases
    {
        /// <summary>
        /// Obtiene todos los estados de pedido mapeados a DTO.
        /// </summary>
        /// <returns>Lista de <see cref="EstadoPedidoDTO"/> ordenada por <c>OrdenEstado</c>.</returns>
        Task<List<EstadoPedidoDTO>> GetAllAsync();

        /// <summary>
        /// Obtiene un estado de pedido por su identificador único mapeado a DTO.
        /// </summary>
        /// <param name="id">Identificador único del estado a buscar.</param>
        /// <returns>El <see cref="EstadoPedidoDTO"/> encontrado, o <c>null</c> si no existe.</returns>
        Task<EstadoPedidoDTO?> GetByIdAsync(int id);
    }
}

File: frontend/src/app/features/pedidos/domain/interfaces/usecases/IEstadoPedidoUseCases.ts
import { EstadoPedido } from '../../entities/estado_pedido';

/**
 * Contrato de los casos de uso de estado de pedido.
 * @remarks Implementado por {@link EstadoPedidoUseCases}.
 */
export interface IEstadoPedidoUseCases {
  /** Devuelve todos los estados de pedido disponibles. */
  getAllAsync(): Promise<EstadoPedido[]>;
  /**
   * Obtiene un estado de pedido por su identificador.
   * @param id Identificador del estado.
   * @returns Entidad {@link EstadoPedido} o `null` si no existe.
   */
  getByIdAsync(id: number): Promise<EstadoPedido | null>;
}
